fastlane/lanes/build.rb

```diff
@@ -397,4 +399,21 @@ def read_version_from_config
       configuration.attributes['VERSION_SHORT']
     end
   end
+
+  def add_buildkite_analytics_token(xctestrun_path:)
+    require 'plist'
+
+    token_key = 'BUILDKITE_ANALYTICS_TOKEN'
+    token = ENV.fetch(token_key)
+    return if token.nil?
+
+    xctestrun = Plist.parse_xml(xctestrun_path)
+    xctestrun['TestConfigurations'].each do |configuration|
+      configuration['TestTargets'].each do |target|
+        target['EnvironmentVariables'][token_key] = token
+      end
+    end
+
+    File.write(xctestrun_path, Plist::Emit.dump(xctestrun))
+  end
```

Comment:
See comment in the matching WooCommerce PR: https://github.com/woocommerce/woocommerce-ios/pull/7097#discussion_r903144984

Reply:
Same concerns as with the WCiOS PR about security and risk of accidentally committing the token if we run the lane locally (as MIEs do for debugging stuff…)